### PR TITLE
added singleton using lazy initialisation method on Ui class

### PR DESCRIPTION
**Pull request by qjie7**

### CODE REVIEW

**Comment by lmtaek** [comment]
Reviewed. Looks good to me.

File: src/main/java/duke/Duke.java [hunk 1]
```diff
@@ -30,15 +30,15 @@
     /**
      * A Ui object that deals with interactions with the user.
      */
-    private Ui ui;
+    private Ui ui = Ui.getUi();
```
Comment (HUANGXUANKUN) [inline src/main/java/duke/Duke.java]: Looks good to me for the UI case

File: src/main/java/duke/Duke.java [hunk 2]
```diff
@@ -30,15 +30,15 @@
...
      * Constructs a Duke object with a relative file path.
      * Initialize the user interface and reads tasks from the specific text file.
      * @param filePath A string that represents the path of the local file
      *          used for storing tasks.
      */
     public Duke(String filePath) {
-        ui = new Ui();
+        //ui = Ui.getUi();
         storage = new Storage(filePath + "/data.txt");
         patientStorage = new PatientStorage(filePath + "/patients.csv");
```
Comment (HUANGXUANKUN) [inline src/main/java/duke/Duke.java]: If the same approach is applied to taskStorage/patientStorage, you will need to define filePath somewhere else before doing this.

Comment (HUANGXUANKUN) [reply src/main/java/duke/Duke.java]: or else we can hardcode the file path too, like the following

` private TaskStorage taskStorage = TaskStorage.getTaskStorage ("./data/data.txt")`

**Pull request merged**